src/Services/CleanupService.php

```diff
@@ -174,90 +175,110 @@ protected function setActiveSlaveGroups(): void
     /**
      * removes everything that is not part of the active
      * groups array
+     * @param int $maxProcesses
      */
-    private function removeInactiveSlaveGroups(): void
+    private function removeInactiveSlaveGroups(int $maxProcesses = null): void
```

Comment:
It's unlikely that `$maxProcesses` is going to change once it's been set once. Prefer to make this a setter to overwrite the amount of processes to spawn, and don't let users overwrite it on the individual functions.